os-eap-launch/added/launch/jgroups.sh

```diff
@@ -22,21 +22,41 @@ function configure_jgroups_encryption() {
   jgroups_encrypt=""
 
   if [ -n "${JGROUPS_ENCRYPT_SECRET}" ]; then
-    if [ -n "${JGROUPS_ENCRYPT_NAME}" -a -n "${JGROUPS_ENCRYPT_PASSWORD}" ] ; then
-      # For new JGroups we need to use SYM_ENCRYPT protocol
-      jgroups_encrypt="\
-        <protocol type=\"SYM_ENCRYPT\">\
-          <property name=\"provider\">SunJCE</property>\
-          <property name=\"sym_algorithm\">AES</property>\
-          <property name=\"encrypt_entire_message\">true</property>\
-          <property name=\"keystore_name\">${JGROUPS_ENCRYPT_KEYSTORE_DIR}/${JGROUPS_ENCRYPT_KEYSTORE}</property>\
-          <property name=\"store_password\">${JGROUPS_ENCRYPT_PASSWORD}</property>\
-          <property name=\"alias\">${JGROUPS_ENCRYPT_NAME}</property>\
-        </protocol>"
-    else
-      log_warning "Partial JGroups encryption configuration, the communication within the cluster WILL NOT be encrypted."
-    fi
+    JGROUPS_ENCRYPT_PROTOCOL="SYM_ENCRYPT"
```

Comment:
Won't this always produce `SYM_ENCRYPT`?  Shouldn't it be `JGROUPS_ENCRYPT_PROTOCOL="${JGROUPS_ENCRYPT_PROTOCOL:-SYM_ENCRYPT}"`?

Reply:
Correct. It will always produce ```SYM_ENCRYPT``` when the ```JGROUPS_ENCRYPT_SECRET``` is also set. The JGroups ASYM_ENCRYPT protocol will be set if and only if:

- ```JGROUPS_ENCRYPT_PROTOCOL``` is set to ```ASYM_ENCRYPT```, and
- None of ```JGROUPS_ENCRYPT_SECRET```, ```JGROUPS_ENCRYPT_NAME```, and ```JGROUPS_ENCRYPT_PASSWORD``` are set. Per JGroups documentation, ASYM_ENCRYPT doesn't need these (the JGroups keystore definition), since it will generate the public / private key pair by itself under the hood.

So the assumption was, if the user provided JGroups keystore, they want ```SYM_ENCRYPT```. When they don't provide it, they want ```ASYM_ENCRYPT```.

Reply:
I think we default it if it's not set.  If they specify both, we just log a warning, e.g. "JGROUPS_ENCRYPT_SECRET ignored when configuring ASYM_ENCRYPT."  This provides a more natural use case, where folks might want to migrate to ASYM_ENCRYPT, but forget to nuke the other variable setting.

Reply:
Fair enough. Your proposal looks reasonable. Will update the PR yet. Thanks for these suggestions!